nettrine/app/Presenters/AdvancedPresenter.php

```diff
@@ -158,10 +158,13 @@ public function actionDeleteArticle(int $id): void
 		$articleRepository = $this->em->getArticleRepository();
 		$article = $articleRepository->find($id);
 
-		$this->em->remove($article);
-		$this->em->flush();
+		if ($article instanceof Article) {
```

Comment:
Takhle to nechceme resit. Musime nejaky jinaci cestu.

Reply:
Do https://github.com/contributte/playground/blob/522e16070c96743b5f06d2d57b095e6e64706399/nettrine/app/Model/Database/Repository/EntityRepository.php prijde neco jako toto:

```php
	public function fetch(int $id, ?int $lockMode = NULL, ?int $lockVersion = NULL): AbstractEntity
	{
		/** @var AbstractEntity|null $entity */
		$entity = $this->find($id, $lockMode, $lockVersion);

		if (!$entity) {
			throw RuntimeException(...);
		}

		return $entity;
	}

	/**
	 * @param mixed[] $criteria
	 * @param mixed[]|null $orderBy
	 */
	public function fetchBy(array $criteria, ?array $orderBy = NULL): AbstractEntity
	{
		/** @var AbstractEntity|null $entity */
		$entity = $this->findOneBy($criteria, $orderBy);

		if (!$entity) {
			throw RuntimeException(...);
		}

		return $entity;
	}
```

Reply:
nechápu, o tu řešíš a vymýšlíš

Reply:
Resim to, ze `if ($article instanceof Article)` sice vyresi problem s phpstanem, ale neni to neco co bychom takhle chteli mit.